### PR TITLE
Implements Google Cloud Storage metadata store layer

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR adds backend and SDK support for using GCS as the metadata store. There will be a separate PR for the UI changes and a small PR for getting the credentials in a format that is accessible by a k8s cluster.

## Related issue number (if any)
ENG 1597

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/cmd/server/handler/connect_integration.go [hunk 1]
```diff
@@ -287,40 +287,60 @@ func validateAirflowConfig(
...
+	case integration.GCS:
+		var c integration.GCSConfig
+		if err := json.Unmarshal(data, &c); err != nil {
+			return err
+		}
+
+		storageConfig, err = convertGCSIntegrationtoStorageConfig(&c)
+		if err != nil {
+			return err
+		}
```
Comment (hsubbaraj-spiral) [inline src/golang/cmd/server/handler/connect_integration.go:343]: Probably add some nil default case here.

File: src/golang/cmd/server/handler/connect_integration.go [hunk 2]
```diff
@@ -402,6 +422,16 @@ func convertS3IntegrationtoStorageConfig(c *integration.S3Config) (*shared.Stora
 	return storageConfig, nil
 }
 
+func convertGCSIntegrationtoStorageConfig(c *integration.GCSConfig) (*shared.StorageConfig, error) {
```
Comment (hsubbaraj-spiral) [inline src/golang/cmd/server/handler/connect_integration.go:425]: should be just `*shared.StorageConfig `as a return type?

**Review by hsubbaraj-spiral — APPROVED** [review]
looks good, just a few nits.

**Review by cw75 — APPROVED** [review]
LGTM!

**Pull request merged**